File: src/main/java/fi/metatavu/metaform/server/files/File.java
package fi.metatavu.metaform.server.files;

/**
 * Class representing a file uploaded into the system but not yet persisted into the database
 * 
 * @author Antti Leppä
 */
public class File {

  private FileMeta meta;
  private byte[] data;

  /**
   * Constructor
   * 
   * @param meta file meta
   * @param data file data
   */
  public File(FileMeta meta, byte[] data) {
    super();
    this.meta = meta;
    this.data = data;
  }
  
  public byte[] getData() {
    return data;
  }

  public FileMeta getMeta() {
    return meta;
  }

}

File: src/main/kotlin/fi/metatavu/metaform/server/files/File.kt
package fi.metatavu.metaform.server.files

data class File (
        var meta: FileMeta,
        var data: ByteArray
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as File

        if (meta != other.meta) return false
        if (!data.contentEquals(other.data)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = meta.hashCode()
        result = 31 * result + data.contentHashCode()
        return result
    }
}